### PR TITLE
Set up Karma and Travis

### PR DESCRIPTION
**Pull request by jeffbcross**

### CODE REVIEW

**Comment by jeffbcross** [comment]
@robwormald care to review this?

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+install:
+  - npm install
+
+before_script:
+  - export DISPLAY=:99.0
+  - sh -e /etc/init.d/xvfb start
+
+script:
+  - ./node_modules/.bin/karma start --single-run --browsers Firefox
```
Comment (robwormald) [inline .travis.yml:25]: Do we want to test against Chrome as well?

Comment (jeffbcross) [inline .travis.yml:25]: Yeah, but Firefox is easier on Travis. I originally set it up with Chrome, but there was a cryptic runtime error that I couldn't reproduce locally on my linux box. I'll open an issue to set up Browser Stack for more browsers.

**Comment by jeffbcross** [comment]
I opened #18 as a follow-up to set up more browsers.

File: karma.conf.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+          }
+        }
+      }
+    },
+    files: [
+      // Just sourcemaps needed for SystemJS since karma plugin provides System
+      {pattern: 'node_modules/systemjs/dist/**/*.map', included: false, watched: false},
+      {pattern: 'node_modules/reflect-metadata/**/*.js', included: false, watched: false},
+      // Include reflect so it will be globally available before Angular loads
+      'node_modules/reflect-metadata/Reflect.ts',
```
Comment (jeffbcross) [inline karma.conf.js:32]: Use angular2-polyfills instead. Should also replace zone.

File: karma.conf.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      // Cannot wholesale include zone because load ordering matters.
+      {pattern: 'node_modules/zone.js/lib/**/*.js', included: false, watched: false},
```
Comment (jeffbcross) [inline karma.conf.js:34]: See line above

File: karma.conf.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+      // Include zone so it will automatically patch.
+      'node_modules/zone.js/lib/zone.js',
+      {pattern: 'node_modules/rxjs/**/*.@(js|map)', included: false, watched: false},
+      {pattern: 'node_modules/firebase/lib/firebase-web.js', included: false, watched: false},
+      {pattern: 'node_modules/angular2/**/*.js', included: false, watched: false},
+
+      {pattern: 'karma-test-shim.js', included: true, watched: true},
```
Comment (jeffbcross) [inline karma.conf.js]: Make string instead of object

File: package.json [hunk 5]
```diff
@@ -9,20 +9,28 @@
...
+    "rxjs": "5.0.0-beta.0",
     "zone.js": "^0.5.8"
   },
   "devDependencies": {
+    "es6-module-loader": "^0.17.10",
     "gulp": "^3.9.0",
     "gulp-inline": "^0.1.0",
     "gulp-minify-css": "^1.2.2",
     "gulp-sass": "^2.1.1",
+    "gulp-typescript": "^2.10.0",
```
Comment (jeffbcross) [inline package.json]: Remove this.

**Comment by jeffbcross** [comment]
- [x] Open up issue to make this artifact-based instead of using preprocessor

**Pull request merged**